.github/workflows/branch.yml

```diff
@@ -0,0 +1,15 @@
+name: Branch CI
+
+on:
+  push:
+    branches-ignore:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```

Comment:
The group here creates an ID for the current CI run right?

Reply:
Yes. With this line we tell Github Actions to only have the latest CI running and abort the previous one. This is especially helpful if you push multiple times in a short time.